frontend/model/contracts/group.js

```diff
@@ -1300,28 +1300,54 @@ sbp('chelonia/defineContract', {
       const { paymentsByPeriod, payments } = archivingPayments
       const { username } = sbp('state/vuex/state').loggedIn
 
-      for (const period of Object.keys(paymentsByPeriod).sort()) {
-        const archPaymentsByPeriodKey = `paymentsByPeriod/${username}/${contractID}`
-        const archPaymentsByPeriod = await sbp('gi.db/archive/load', archPaymentsByPeriodKey) || {}
-        const archPaymentsKey = `payments/${username}/${contractID}`
-        let archPayments = await sbp('gi.db/archive/load', archPaymentsKey) || {}
+      const archPaymentsByPeriodKey = `paymentsByPeriod/${username}/${contractID}`
+      const archPaymentsByPeriod = await sbp('gi.db/archive/load', archPaymentsByPeriodKey) || {}
+      const archSentOrReceivedPaymentsKey = `sentOrReceivedPayments/${username}/${contractID}`
+      const archSentOrReceivedPayments = await sbp('gi.db/archive/load', archSentOrReceivedPaymentsKey) || { sent: [], received: [] }
+
+      const sortPayments = payments => payments.sort((f, l) => f.meta.createdDate < l.meta.createdDate ? 1 : -1)
 
+      for (const period of Object.keys(paymentsByPeriod).sort()) {
```

Comment:
This function is very large and complicated and difficult to follow. It is made all the more difficult to follow by the lack of comments.

Could you improve its readability by using single-line comments above important sections and blocks explaining what they're about to do?

For example, a comment above this `for` loop could say something like this:

```js
// add the new archived payment data to the collection of all archived send and received payments
```

Good comments explain conceptually what's going on and what is the goal that we're trying to achieve.